lib/is.js

```diff
@@ -71,6 +71,81 @@ var util = require('./util');
     });
 })();
 
+/**
+ * Steam tests.
+ */
+(function () {
+    var identifier = '[0-9\\-_]{1,100}';
+    def({
+        fn: 'steam.id',
+        re: new RegExp('^' + identifier + '$', 'i')
+    });
+    def({
+        fn: 'steam.profile',
+        re: url('steamcommunity\\.com\\/profiles\\/.+' + identifier + '\\/')
+    });
+})();
+
+/**
+ * Instagram tests.
+ */
+(function () {
+    var identifier = '[a-z0-9\\-_]{1,30}';
+    def({
+        fn: 'instagram.name',
+        re: new RegExp('^' + identifier + '$', 'i')
+    });
+    def({
+        fn: 'instagram.url',
+        re: url('instagram\\.com\\/.+' + identifier)
+    });
+})();
+
+/**
+ * SoundCloud tests.
+ */
+(function () {
+    var identifier = '[a-z0-9\\-_]{1,100}';
+    def({
+        fn: 'soundcloud.name',
+        re: new RegExp('^' + identifier + '$', 'i')
+    });
+    def({
+        fn: 'soundcloud.url',
+        re: url('soundcloud\\.com\\/.+' + identifier)
+    });
+})();
+
+/**
+ * Spreadshirt tests.
+ */
+(function () {
+    var identifier = '[a-z0-9\\-_]{1,100}';
+    def({
+        fn: 'spreadshirt.name',
+        re: new RegExp('^' + identifier + '$', 'i')
+    });
+    def({
+        fn: 'spreadshirt.url',
+        re: url('spreadshirt\\.com\\/user\\/.+' + identifier)
+    });
+})();
+
+/**
+ * Patreon tests.
+ */
+(function () {
+    var identifier = '[a-z0-9\\-_]{1,100}';
+    def({
+        fn: 'patreon.name',
+        re: new RegExp('^' + identifier + '$', 'i')
+    });
+    def({
+        fn: 'patreon.url',
+        re: url('patreon\\.com\\/.+' + identifier)
```

Comment:
Ending these with `.+[a-z0-9\\-_]{1,100}` is probably not what you want. It'll match anything that ends in a-z0-9-_, and is backtracking